src/main/java/com/google/sps/servlets/AggregationServlet.java

```diff
@@ -75,10 +75,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     return allDevices;
   }
 
-  public static MultiKeyMap<String, List<String>> processData(
+  public static MultiKeyMap<String, List<ChromeOSDevice>> processData(
       List<ChromeOSDevice> devices,
       LinkedHashSet<AnnotatedField> fields) {
-    MultiKeyMap<String, List<String>> counts = new MultiKeyMap<>();
+    MultiKeyMap<String, List<ChromeOSDevice>> aggregationEntries = new MultiKeyMap<>();
```

Comment:
in general, we could make certain objects final in order to avoid accidentally changing their reference

Reply:
I've changed the function signature to make the arguments `final`, as those should definitely not be changed.  However I feel like making all local variables `final` is a bit overkill and just adds more visual clutter to sift through when reading the code.  I've looked through the google style guide and it seems like there is no general prescription on which way is preferred (it seems like use of `final` for local variables is generally decided on a project-by-project basis).  For now I am going to leave the local variables as-is, and perhaps we can decide on a standard for the project at a later date.

Reply:
While I agree that it may be overkill at times and introduce visual clutter, I believe correctness is among the most desirable properties, one that is much more important personally than code visual elegance. 

Especially because our code test coverage is somewhat lacking, adding a few compile-time checks with the use of final would make our servlets much more resistant to certain bugs.